plugin/pkg/admission/serviceplan/defaultserviceplan/admission.go

```diff
@@ -134,6 +146,67 @@ func (d *defaultServicePlan) Admit(a admission.Attributes) error {
 	return nil
 }
 
+func (d *defaultServicePlan) handleDefaultServicePlan(a admission.Attributes, instance *servicecatalog.ServiceInstance) error {
+	// Need to dynamically set the namespaced clients based on the namespace of the instance
+	d.scClient = d.internalClientSet.Servicecatalog().ServiceClasses(instance.Namespace)
+	d.spClient = d.internalClientSet.Servicecatalog().ServicePlans(instance.Namespace)
+
+	sc, err := d.getServiceClassByPlanReference(a, &instance.Spec.PlanReference)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return admission.NewForbidden(a, err)
+		}
+		msg := fmt.Sprintf("ServiceClass %c does not exist, can not figure out the default ServicePlan.",
+			instance.Spec.PlanReference)
+		glog.V(4).Info(msg)
+		return admission.NewForbidden(a, errors.New(msg))
+	}
+
+	// find all the service plans that belong to the service class
+
+	// Need to be careful here. Is it possible to have only one
+	// ServicePlan available while others are still in progress?
+	// Not currently. Creation of all ServicePlans before creating
+	// the ServiceClass ensures that this will work correctly. If
+	// the order changes, we will need to rethink the
+	// implementation of this controller.
+	plans, err := d.getServicePlansByServiceClassName(sc.Name)
+	if err != nil {
+		msg := fmt.Sprintf("Error listing ServicePlans for ServiceClass (K8S: %v ExternalName: %v) - retry and specify desired ServicePlan", sc.Name, sc.Spec.ExternalName)
+		glog.V(4).Infof(`ServiceInstance "%s/%s": %s`, instance.Namespace, instance.Name, msg)
+		return admission.NewForbidden(a, errors.New(msg))
+	}
+
+	// check if there were any service plans
+	// TODO: in combination with not allowing classes with no plans, this should be impossible
+	if len(plans) <= 0 {
+		msg := fmt.Sprintf("no ServicePlans found at all for ServiceClass %q", sc.Spec.ExternalName)
+		glog.V(4).Infof(`ServiceInstance "%s/%s": %s`, instance.Namespace, instance.Name, msg)
+		return admission.NewForbidden(a, errors.New(msg))
+	}
+
+	// check if more than one service plan was specified and error
```

Comment:
s/specified/found/

user didn't specify a plan, we are trying to locate a default plan